File: code/1.two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const numMap: Map<number, number> = new Map();

    for (const [i, n] of nums.entries()) {
        const diff = target - n;
        const diffIndex = numMap.get(diff);

        if (diffIndex != undefined) return [diffIndex, i];

        numMap.set(n, i);
    }

    return [];
}

File: code/1.two-sum.go
func twoSum(nums []int, target int) []int {
	nMap := make(map[int]int)

	for i, n := range nums {
		diff := target - n
		diffIndex, ok := nMap[diff]

		if ok {
			return []int{diffIndex, i}
		}

		nMap[n] = i
	}

	return []int{}
}
